python/pyarrow/src/arrow/python/udf.cc

```diff
@@ -83,16 +135,17 @@ struct PythonUdf : public compute::KernelState {
   }
 };
 
-Status PythonUdfExec(compute::KernelContext* ctx, const ExecSpan& batch,
-                     ExecResult* out) {
+Status PythonUdfExec(compute::KernelContext* ctx, const compute::ExecSpan& batch,
+                     compute::ExecResult* out) {
   auto udf = static_cast<PythonUdf*>(ctx->kernel()->data.get());
   return SafeCallIntoPython([&]() -> Status { return udf->Exec(ctx, batch, out); });
 }
 
-}  // namespace
-
-Status RegisterScalarFunction(PyObject* user_function, ScalarUdfWrapperCallback wrapper,
-                              const ScalarUdfOptions& options) {
+Status RegisterScalarLikeFunction(PyObject* user_function,
```

Comment:
I think this PR muddles the meaning of the word "scalar" which is already a confusing enough word as it is.  Do we want to maybe pick a new word entirely instead of "scalar like"?  We should also rename ScalarUdfWrapperCallback, ScalarUdfOptions as well (at the very least to ScalarLike).  What would you imagine is not a "scalar like" function?  The pattern (something with a python function object and options that describe the behavior of that function object) seems pretty generic.

If we can't think of anything that wouldn't fit this pattern then maybe "PythonFunction", "PythonUdfWrapperCallback", and "PythonUdfOptions"?

Or maybe, compared to aggregate udfs, the unique thing is that there is one function.  So maybe "SingleFunction", "SingleFunctionUdfWrapperCallback", and "SingleFunctionUdfOptions"?

Reply:
That sounds good to me.  We can worry about distinguishing when we have a case that doesn't fit the pattern.